polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -5107,7 +5174,8 @@ void MLIRASTConsumer::HandleDeclContext(DeclContext *DC) {
     if ((emitIfFound.count("*") && name != "fpclassify" && !fd->isStatic() &&
          externLinkage) ||
         emitIfFound.count(name)) {
-      functionsToEmit.push_back(fd);
+      // Functions are created in the host context by default.
+      functionsToEmit.emplace_back(fd, FunctionContext::Host);
```

Comment:
I don't think this is going to work because there are cases when we want to create a function in the device context.

Reply:
We cannot know the context, as the function being declared might be called from any context. I'm creating it in the host context by default and the one in the device context will be created only if we actually find a call for it in device code.

Reply:
for now is fine